lib/ts/recipe/webauthn/types.ts

```diff
@@ -0,0 +1,625 @@
+            | {
+                  session: SessionContainerInterface;
+              }
+        )
+    ): Promise<
+        | {
+              status: "OK";
+              webauthnGeneratedOptionsId: string;
+              rp: {
+                  id: string;
+                  name: string;
+              };
+              user: {
+                  id: string;
+                  name: string; // user email
+                  displayName: string; //user email
+              };
+              challenge: string;
+              timeout: number;
+              excludeCredentials: {
+                  id: string;
+                  type: "public-key";
+                  transports: ("ble" | "hybrid" | "internal" | "nfc" | "usb")[];
+              }[];
+              attestation: "none" | "indirect" | "direct" | "enterprise";
+              pubKeyCredParams: {
+                  // we will default to [-8, -7, -257] as supported algorithms. See https://www.iana.org/assignments/cose/cose.xhtml#algorithms
+                  alg: number;
+                  type: "public-key";
+              }[];
+              authenticatorSelection: {
+                  requireResidentKey: boolean;
+                  residentKey: "required" | "preferred" | "discouraged";
+                  userVerification: "required" | "preferred" | "discouraged";
+              };
+          }
+        | { status: "EMAIL_MISSING_ERROR" } // email is required if not using session or recoverAccountToken
+        | { status: "SESSION_MISSING_ERROR" } // session is required if not using email or recoverAccountToken
+        | { status: "RECOVER_ACCOUNT_TOKEN_MISSING_ERROR" } // recoverAccountToken is required if not using email or session
+        | { status: "RECOVER_ACCOUNT_TOKEN_INVALID_ERROR" }
+        | { status: "RELYING_PARTY_ID_MISSING_ERROR" }
+        | { status: "RELYING_PARTY_ID_INVALID_ERROR" } // wrong format; must be domain
+        | { status: "RELYING_PARTY_NAME_MISSING_ERROR" } // relyingPartyName is required
+        | { status: "ORIGIN_MISSING_ERROR" }
+        | { status: "ORIGIN_INVALID_ERROR" } // wrong format; must be domain url
+    >;
+
+    signInOptions(input: {
+        relyingPartyId: string;
+        origin: string;
+        userVerification: "required" | "preferred" | "discouraged" | undefined; // see register options
+        timeout: number | undefined;
+        tenantId: string;
+        userContext: UserContext;
+    }): Promise<
+        | {
+              status: "OK";
+              webauthnGeneratedOptionsId: string;
+              challenge: string;
+              timeout: number;
+              userVerification: "required" | "preferred" | "discouraged";
+          }
+        | { status: "RELYING_PARTY_ID_MISSING_ERROR" }
+        | { status: "RELYING_PARTY_ID_INVALID_ERROR" } // wrong format; must be domain
+        | { status: "ORIGIN_MISSING_ERROR" }
+        | { status: "ORIGIN_INVALID_ERROR" } // wrong format; must be domain url
```

Comment:
see comments for registerOptions output please.

Reply:
done. will be an exception